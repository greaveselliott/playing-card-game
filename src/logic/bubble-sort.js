const bubble_sort = (array, key) => {
    var swapped;
    do {
        swapped = false;
        for (var i=0; i < array.length-1; i++) {

            if (array[i][key] > array[i+1][key]) {
                var cached_index = array[i];
                array[i] = array[i+1];
                array[i+1] = cached_index;
                swapped = true;
            }
        }
    } while (swapped);

    return array;
}

export default bubble_sort;