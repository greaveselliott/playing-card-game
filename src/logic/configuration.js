export const card_suits = ["Clubs", "Hearts", "Diamonds", "Spades"];
export const card_numbers = ["2", "3", "4", "5", "6", "7", "8", "9", "10", "J", "Q","K","A"];

export const deck_factory = (card_suits, card_numbers) => {
    let deck = [];

    card_suits.forEach((suit) => {
        card_numbers.forEach((number) => {
            deck.push({
                id: deck.length,
                suit,
                number
            })
        });
    });

    return deck;
};