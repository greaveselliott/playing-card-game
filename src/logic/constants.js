export const DRAW_INCREMENT = "DRAW_INCREMENT";
export const DRAW_DECREMENT = "DRAW_DECREMENT";
export const DRAW_CARDS = "DRAW_CARDS";
export const SORT_CARDS = "SORT_CARDS";
export const SHUFFLE_DECK = "SHUFFLE_DECK";