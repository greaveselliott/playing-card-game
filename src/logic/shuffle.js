const shuffle = (array) => {
  var copy = [], n = array.length, i;

  while (n) {
    i = Math.floor(Math.random() * n--);
    
    copy.push(array.splice(i, 1)[0]);
  }

  return copy;
}

export default shuffle;