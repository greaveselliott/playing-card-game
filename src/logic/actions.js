import * as CONSTANTS from './constants';

export const draw_increment = () => {
    return {
        type: CONSTANTS.DRAW_INCREMENT
    }
};

export const draw_decrement = () => {
    return {
        type: CONSTANTS.DRAW_DECREMENT
    }
};

export const draw_cards = () => {
    return {
        type: CONSTANTS.DRAW_CARDS
    }
};

export const sort_cards = () => {
    return {
        type: CONSTANTS.SORT_CARDS
    }
};

export const shuffle_deck = () => {
    return {
        type: CONSTANTS.SHUFFLE_DECK
    }
};

export const shuffle_hand = () => {
    return {
        type: CONSTANTS.SHUFFLE_HAND
    }
};