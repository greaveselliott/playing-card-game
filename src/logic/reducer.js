import * as CONSTANTS from './constants';
import _ from 'lodash';
import * as configuration from './configuration';

export const initial_state = {

};

const reducers = (state = initial_state, action) => {
  switch (_.get(action, 'type', undefined)) {


    default:
      return state;
  }
};

export default reducers;
