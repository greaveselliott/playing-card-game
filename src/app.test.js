import React from 'react';
import App from './app';
import { createMockStore } from 'redux-test-utils';
import { shallowWithStore } from 'enzyme-redux';
import { initial_state } from './logic/reducer';

describe("App container", () => {  

    it("Should map dispatch methods to control props.", () => {

        const mockStore = createMockStore({ initial_state });

        const component = shallowWithStore(<App />, mockStore);

        expect(component).toMatchSnapshot();
    });
});