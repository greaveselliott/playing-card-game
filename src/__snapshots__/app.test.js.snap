// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container Should map dispatch methods to control props. 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Provider
    store={
        Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
    }
>
    <div
        className="l-app"
    >
        <main
            className="l-app__main"
        >
            <Connect(CardCollection)
                heading="Deck"
            />
            <Connect(CardCollection)
                heading="Hand"
            />
            <Connect(Controls) />
        </main>
    </div>
</Provider>,
  "nodes": Array [
    <Provider
      store={
            Object {
                  "dispatch": [Function],
                  "getState": [Function],
                  "replaceReducer": [Function],
                  "subscribe": [Function],
                  Symbol(observable): [Function],
                }
      }
>
      <div
            className="l-app"
      >
            <main
                  className="l-app__main"
            >
                  <Connect(CardCollection)
                        heading="Deck"
                  />
                  <Connect(CardCollection)
                        heading="Hand"
                  />
                  <Connect(Controls) />
            </main>
      </div>
</Provider>,
  ],
  "options": Object {
    "context": Object {
      "store": Object {
        "dispatch": [Function],
        "getAction": [Function],
        "getActions": [Function],
        "getState": [Function],
        "isActionDispatched": [Function],
        "isActionTypeDispatched": [Function],
        "subscribe": [Function],
      },
    },
  },
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {
        "store": Object {
          "dispatch": [Function],
          "getAction": [Function],
          "getActions": [Function],
          "getState": [Function],
          "isActionDispatched": [Function],
          "isActionTypeDispatched": [Function],
          "subscribe": [Function],
        },
      },
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <div
                    className="l-app"
          >
                    <main
                              className="l-app__main"
                    >
                              <Connect(CardCollection)
                                        heading="Deck"
                              />
                              <Connect(CardCollection)
                                        heading="Hand"
                              />
                              <Connect(Controls) />
                    </main>
          </div>
</Provider>,
        "_debugID": 2,
        "_renderedOutput": <Provider
          store={
                    Object {
                              "dispatch": [Function],
                              "getState": [Function],
                              "replaceReducer": [Function],
                              "subscribe": [Function],
                              Symbol(observable): [Function],
                            }
          }
>
          <div
                    className="l-app"
          >
                    <main
                              className="l-app__main"
                    >
                              <Connect(CardCollection)
                                        heading="Deck"
                              />
                              <Connect(CardCollection)
                                        heading="Hand"
                              />
                              <Connect(Controls) />
                    </main>
          </div>
</Provider>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
